pylint/checkers/unicode.py

```diff
@@ -0,0 +1,522 @@
+# Copyright (c) 2021-2022 Carli Freudenberg <carli.freudenberg@energymeteo.de>
+
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+"""Unicode and some other ASCII characters can be used to create programs that run
+much different compared to what a human reader would expect from them.
+
+PEP 672 lists some examples.
+See: https://www.python.org/dev/peps/pep-0672/
+
+The following checkers are intended to make users are aware of these issues.
+"""
+import codecs
+import contextlib
+import io
+import re
+from collections import OrderedDict
+from functools import lru_cache
+from tokenize import detect_encoding
+from typing import Dict, Iterable, NamedTuple, Tuple, TypeVar
+
+from astroid import nodes
+
+from pylint import interfaces
+from pylint.checkers import BaseChecker
```

Comment:
This can create circular imports easily if we add the UnicodeChecker  to the checkers API, it should import from the file directly.

Reply:
Changed